packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -328,5 +370,7 @@ else
       EmbedFlutterFrameworks ;;
     "embed_and_thin")
       EmbedAndThinFrameworks ;;
+    "observatory_bonjour_service")
+      AddObservatoryBonjourService ;;
```

Comment:
This is a new xcode_backend entrypoint.  At the least, we can tell add-to-app customers to call this in a new build phase.

Reply:
I think I would rather not add this for add-to-app.

Those customers should be sophisticated enough to read an error log and know which plist(s) to update.

This is likely to get things wrong for some customer and we'll get requests to change it or a specific CI system or wahtever.

Reply:
What about .frameworks? Will those cases never work unless users type in something manually in their real checked-in info.plist?

Reply:
> Those customers should be sophisticated enough to read an error log and know which plist(s) to update.

How would they see an error message from the VM if they can't attach?
It definitely requires some Xcode-foo to know you can edit the Info.plist per configuration.  You need to know how to expand the build setting, which people seem confused by.  
<img width="862" alt="Screen Shot 2020-09-01 at 12 16 17 AM" src="https://user-images.githubusercontent.com/682784/91814041-66ed7700-ebe8-11ea-93c1-912feed22d9e.png">
Or they can do something like `Info-${CONFIGURATION}.plist`
We'll have to update the docs, for sure.

>  Will those cases never work unless users type in something manually in their real checked-in info.plist?

Right, they would need to edit their plist.

At least in the CocoaPods case they can add the build phase I suggested in https://github.com/flutter/flutter/issues/63893#issuecomment-684126922